src/samVisualize/graphGeneration/cfnTemplateGraphGenerator.ts

```diff
@@ -0,0 +1,135 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Graph, GraphObject } from './graph'
+import { yamlParse } from 'yaml-cfn'
+import { LinkTypes } from '../samVisualizeTypes'
+import * as _ from 'lodash'
+import { getLogger } from '../../shared/logger/logger'
+
+/**
+ * @param inputYaml A string representing a YAML template
+ * @returns A JavaScript object corresponding to the given YAML string
+ */
+function yamlStringToObject(inputYaml: string): Record<string, unknown> {
+    let templateAsObject = {}
+    try {
+        templateAsObject = yamlParse(inputYaml)
+    } catch (err) {
+        getLogger().error(
+            `SAM Visualize: Failed to load template "${inputYaml.substr(
+                0,
+                25
+            )}"... Please ensure the template is valid YAML: 0%`,
+            err
+        )
+    }
+    return templateAsObject
```

Comment:
This will always return. Is this intended?

Reply:
It's better to return `undefined` in the event of a parsing error than an empty object, so I'll fix this